### PR TITLE
feature/config-bulk-import-async

### PR DESCRIPTION
**Pull request by jrouzierinverse**
NOT READY
=========

Description
===========
Allow the bulk import of config items to be async.

Impacts
=======
Bulk config import

NEWS file entries
=======================
Enhancements
-------------
* Allow the bulk import of config items to be async.

Issue
=====
Related #5920

Delete branch after merging.
=========================
NO

Checklist
=========
- [x] Add unit tests

### CODE REVIEW

**Comment by jrouzierinverse** [comment]
@cgx @satkunas 
This can be used by using setting the async flags.
```json
{ "async" : true, "items" : [ {  "id" : "value_1" , "value_2" : "2" } ] }
```
Results
```json
{ "task_id" : "54321" }
```

To get the results use the task poll API

GET /api/v1/pfqueue/task/54321/status/poll

**Comment by jrouzierinverse** [comment]
Can this be merged and an issue create to implement this logic in the frontend.

**Comment by satkunas** [comment]
@jrouzierinverse frontend pushed, ready to merge

**Comment by julsemaan** [comment]
@satkunas, does this need to be revisited with the admin rework?

If so, this needs to be postponed

**Comment by satkunas** [comment]
@julsemaan I resolved a small merge conflict. I don't see anything else standing in the way from being merged.

**Comment by extrafu** [comment]
@satkunas Make sure it's still relevant.

**Comment by satkunas** [comment]
UI code is OK to merge

File: lib/pf/UnifiedApi/Controller/Config.pm [hunk 1]
```diff
@@ -1300,10 +1305,39 @@ sub bulk_import {
...
+                $updater->completed({items => $results});
+                return;
+            },
+            sub { } # Do nothing
+        );
+
+        return $self->render( json => {status => 202, task_id => $task_id }, status => 202);
+    }
+
+    my $results = $self->do_bulk_import($data);
```
Comment (julsemaan) [inline lib/pf/UnifiedApi/Controller/Config.pm]: An else statement would make things clearer that this is the synchronous import that will only be called if async is false

**Review by julsemaan — CHANGES_REQUESTED** [review]

**Pull request merged**